### ISSUE TITLE
tools.download and redirections

### ISSUE THREAD

**Issue by theodelrieu** [conan-io/conan#2063]
Hi,

I didn't see an equivalent to the curl `-C` option in the docs, I wanted to download the JDK directly from the oracle website, but simply adding the required headers was not enough.

Do you think it would be reasonable to add a parameter to `tools.download`?

**Comment by memsharded** [comment]
Is that uppercase C for "continuing", not lowercase c for "cookies", right?
What is the issue with the JDK, and how such a download would be continued? Download in two different calls? Thanks for the feedback!

**Comment by theodelrieu** [comment]
Sorry it's the wrong option, I was talking about the `-L` one.

I used this [script](https://gist.github.com/P7h/9741922#file-jdk_download-sh) and removing the `-L` recreates the issue.

**Comment by lasote** [comment]
is it not just a follow the redirect header? Or what is the case with the jdk page?

**Comment by lasote** [comment]
I think requests library is following these redirects by default. Are you sure that this is the problem? That script is setting some cookie and doing some more things to cheat it.

**Comment by theodelrieu** [comment]
Indeed, I've used the cookie that is present in the script, I'll try again tomorrow morning to provide better feedback.

**Comment by theodelrieu** [comment]
I must have confused the files that I got from using cURL and `tools.download`, the problem is with the cookie:

```python
tools.download("http://download.oracle.com/otn-pub/java/jdk/9.0.1+11/jdk-9.0.1_osx-x64_bin.dmg", "lol.dmg", headers={"Cookie": "oraclelicense=accept-securebackup-cookie"})
```

This fails to download the JDK, the following cURL command line gives the same output:

```shell
$ curl -LR#O http://download.oracle.com/otn-pub/java/jdk/9.0.1+11/jdk-9.0.1_osx-x64_bin.dmg
```

Adding the same cookie solves the issue:

```shell
$ curl -LR#OH "Cookie: oraclelicense=accept-securebackup-cookie"  http://download.oracle.com/otn-pub/java/jdk/9.0.1+11/jdk-9.0.1_osx-x64_bin.dmg
```

**Comment by lasote** [comment]
Could be a problem of the header of the cookie getting lost in the redirect. If you could try using directly the requests python module and it works with the cookies parameter maybe we could expose it to the Conan tools.download()

http://docs.python-requests.org/en/master/api/

If you can't try don't worry, I will, but it could require some time.

Thanks for reporting!

**Comment by theodelrieu** [comment]
I tried to use requests and give a `cookies` dictionary, but I've been unable to download anything yet.

At least the http header is set. It's not a blocking issue anyway.

**Comment by solvingj** [comment]
@theodelrieu FYI we have a Conan package which gets the Azul Systems' certified OpenJDK build instead of Oracle's for a few reasons, one of which is the frustrating process of automating the download from Oracle as you have experienced.  Licensing is also better.   Here's our java_installer package, we're using for bazel. 

https://bintray.com/bincrafters/public-conan/java_installer%3Abincrafters

Hope you find it useful.

**Comment by theodelrieu** [comment]
It works great, thanks a lot!
